addon/-private/system/store.js

```diff
@@ -737,57 +737,50 @@ Store = Service.extend({
     @return {Promise} promise
    */
-  // TODO rename this to have an underscore
-  fetchRecord(internalModel, options) {
-    var typeClass = internalModel.type;
-    var id = internalModel.id;
-    var adapter = this.adapterFor(typeClass.modelName);
+  _fetchRecord(internalModel, options) {
+    let modelClass = internalModel.type;
+    let id = internalModel.id;
+    let adapter = this.adapterFor(modelClass.modelName);
 
-    assert("You tried to find a record but you have no adapter (for " + typeClass + ")", adapter);
-    assert("You tried to find a record but your adapter (for " + typeClass + ") does not implement 'findRecord'", typeof adapter.findRecord === 'function' || typeof adapter.find === 'function');
+    assert("You tried to find a record but you have no adapter (for " + modelClass.modelName + ")", adapter);
+    assert("You tried to find a record but your adapter (for " + modelClass.modelName + ") does not implement 'findRecord'", typeof adapter.findRecord === 'function' || typeof adapter.find === 'function');
 
-    var promise = _find(adapter, this, typeClass, id, internalModel, options);
-    return promise;
+    return _find(adapter, this, modelClass, id, internalModel, options);
   },
 
-  scheduleFetchMany(records) {
-    let internalModels = new Array(records.length);
-    let fetches = new Array(records.length);
-    for (let i = 0; i < records.length; i++) {
-      internalModels[i] = records[i]._internalModel;
-    }
+  _scheduleFetchMany(internalModels) {
+    let fetches = new Array(internalModels.length);
 
     for (let i = 0; i < internalModels.length; i++) {
-      fetches[i] = this.scheduleFetch(internalModels[i]);
+      fetches[i] = this._scheduleFetch(internalModels[i]);
     }
 
-    return Ember.RSVP.Promise.all(fetches);
+    return Promise.all(fetches);
   },
 
-  scheduleFetch(internalModel, options) {
-    var typeClass = internalModel.type;
-
+  _scheduleFetch(internalModel, options) {
     if (internalModel._loadingPromise) { return internalModel._loadingPromise; }
 
-    var resolver = Ember.RSVP.defer('Fetching ' + typeClass + 'with id: ' + internalModel.id);
-    var pendingFetchItem = {
-      record: internalModel,
-      resolver: resolver,
-      options: options
+    let modelClass = internalModel.type;
+    let resolver = Ember.RSVP.defer('Fetching ' + modelClass.modelName + 'with id: ' + internalModel.id);
+    let pendingFetchItem = {
+      internalModel,
+      resolver,
+      options
     };
-    var promise = resolver.promise;
+    let promise = resolver.promise;
 
     internalModel.loadingData(promise);
 
-    if (!this._pendingFetch.get(typeClass)) {
-      this._pendingFetch.set(typeClass, [pendingFetchItem]);
+    if (!this._pendingFetch.get(modelClass)) {
+      this._pendingFetch.set(modelClass, [pendingFetchItem]);
     } else {
-      this._pendingFetch.get(typeClass).push(pendingFetchItem);
+      this._pendingFetch.get(modelClass).push(pendingFetchItem);
```

Comment:
Now that we're using `MapWithDefault`, this can be shortened to `this._pendingFetch.get(modelClass).push(pendingFetchItem)`.